packages/context/src/manifest.ts

```diff
@@ -0,0 +1,227 @@
+import { map } from "lodash";
+import { sep } from "path";
+import { readFile } from "fs-extra";
+import { URI } from "vscode-uri";
+import globby from "globby";
+import { FileChangeType } from "vscode-languageserver";
+import { ManifestDetails } from "./types";
+import { Manifest, FileName } from "@sap-ux/project-access";
+import findUp from "find-up";
+import { findAppRoot } from "./utils";
+import { cache } from "./cache";
+
+export async function readManifestFile(
+  manifestUri: string
+): Promise<Manifest | undefined> {
+  const manifestContent = await readFile(
```

Comment:
Reading file may fail with exception which isn't captured with try block. Consumer should take care. Better to have a unified approach and reject promise with corresponding reason in all potential failures instead of returning resolved to undefined promise.

Reply:
wrap it in `try-cache`. On exception, log trace with a message. Regarding promise rejection, as said, should be done in a separate PR as per need.